source/client-side-encryption/client-side-encryption.rst

```diff
@@ -515,9 +517,43 @@ accept arbitrary strings at runtime for forward-compatibility.
       endpoint: string;
    };
 
-See `Why are extraOptions and kmsProviders maps?`_
 
-Drivers MUST enable TLS for all KMS connections.
+Automatic AWS Credentials
+`````````````````````````
+
+.. versionadded:: 1.9.0 2022/06/22
+
+If the ``aws`` provider property of a KMSProviders_ is present and is an empty
+map/object, the driver MUST be able to populate an AWSKMSOptions_ object
+on-demand if-and-only-if AWS credentials are needed.
+
+.. note:: Drivers SHOULD NOT eagerly fill an empty ``aws`` map property.
```

Comment:
```suggestion
.. note:: Drivers MUST NOT eagerly fill an empty ``aws`` map property.
```

Suggest making this a MUST for consistency among driver implementations. That also agrees with https://github.com/mongodb/specifications#when-in-doubt-use-must-not-should-in-specs